src/dash-table/components/Table/Table.less

```diff
@@ -140,6 +140,69 @@
     }
 }
 
+.previous-next-container {
+    display: inline-block;
+    float: right;
+    padding: 5px 0px;
+}
+
+button.previous-page, button.next-page, .current-page {
+    display: inline-block;
+    margin-left: 5px;
+    margin-right: 5px;
+}
+
+.page-number {
+    font-family: monospace;
+    display: inline-block;
+}
+
+input.current-page {
+    border-bottom: solid lightgrey 1px !important;
+}
+
+input.current-page, input.current-page::placeholder {
+    color: black;
+    border: none;
+    width:30px;
+    text-align:center;
+    font-family: monospace;
+    font-size: 10pt;
+    margin: 0.5em;
+}
+
+.page-number > .last-page {
+    min-width: 30px;
+    display: inline-block;
+    text-align: center;
+    margin: 0.5em;
+}
+
+.last-page {
+    width: 30px;
+    display: inline-block;
+}
+
+input.current-page:focus {
+    outline: none;
+}
+input.current-page:focus::placeholder {
+    opacity: 0;
+}
+
+button.previous-page, button.next-page, button.first-page, button.last-page {
+    transition-duration: 400ms;
+    padding: 5px;
+    border: none;
+}
+
+button.previous-page:hover, button.next-page:hover, button.first-page:hover, button.last-page:hover {
+    color: hotpink;
+}
+button.previous-page:hover:disabled, button.next-page:hover:disabled, button.first-page:hover:disabled, button.last-page:hover:disabled {
+    color: graytext
+}
+
```

Comment:
I suggest adding a class on the parent `div` at https://github.com/plotly/dash-table/blob/dev/src/dash-table/components/ControlledTable/index.tsx#L821 and using that class to wrap both the pagination styling and the `.dash-spreadsheet-container` below to prevent leaking styles outside the table's scope.